src/test/regress/input/multi_outer_join.source

```diff
@@ -113,6 +113,14 @@ WHERE
 	r_custkey = 5;
 
 
+-- Apply a filter before the join
+SELECT
+	count(l_custkey), count(r_custkey)
+FROM
+	multi_outer_join_left a LEFT JOIN multi_outer_join_right b
+	ON (l_custkey = r_custkey AND r_custkey = 5);
+
```

Comment:
I think it'd be a good idea to have a test-case here, where the r_custkey restriction prevents all matches within the left join.